WMFComponents/Sources/WMFComponents/Components/Activity Tab/WMFActivityTabViewModel.swift

```diff
@@ -37,28 +53,54 @@ import WMFData
         localizedStrings.totalHoursMinutesRead(hoursRead, minutesRead)
     }
 
-    public func updateHoursMinutesRead(hours: Int, minutes: Int) {
+    // External updates for async
+    
+    private func updateHoursMinutesRead(hours: Int, minutes: Int) {
         self.hoursRead = hours
         self.minutesRead = minutes
     }
     
     public func updateUsername(username: String) {
         self.username = username
     }
+    
+    private func updateTotalArticlesRead(totalArticlesRead: Int) {
+        self.totalArticlesRead = totalArticlesRead
+    }
+    
+    public func updateDateTimeRead(dateTime: Date) {
+        let calendar = Calendar.current
+        let dateFormatter = DateFormatter()
+        
+        if calendar.isDateInToday(dateTime) {
+            dateFormatter.timeStyle = .short
+            dateFormatter.dateStyle = .none
+        } else {
+            dateFormatter.dateFormat = "MMMM d"
+        }
```

Comment:
We have a DateFormatter extension [file](https://github.com/wikimedia/wikipedia-ios/blob/main/WMFComponents/Sources/WMFComponents/Extensions/DateFormatter%2BExtensions.swift) in WMFComponents - can you put this logic in there for the sake of consistency? It also might give us a performance boost.